File: src/main/java/com/kmarzecki/communicator/security/AuthenticationService.java
package com.kmarzecki.communicator.security;

import com.kmarzecki.communicator.model.Language;
import com.kmarzecki.communicator.model.auth.LoginDto;
import com.kmarzecki.communicator.model.auth.LoginResponse;
import com.kmarzecki.communicator.model.auth.RegisterDto;
import com.kmarzecki.communicator.model.auth.UserResponse;

import java.security.Principal;

/**
 * Authentication service
 */
public interface AuthenticationService {
    /**
     * Login user
     * @param dto dto containing login information
     * @return logged user response
     */
    LoginResponse login(LoginDto dto);

    /**
     * Register user
     * @param registerDto dto containing registration information
     */
    void register(RegisterDto registerDto);

    /**
     * Get user information
     * @param principal user principal
     * @return User information response
     */
    UserResponse getMe(Principal principal);
}

File: src/main/kotlin/com/kmarzecki/trollo/security/AuthenticationService.kt
package com.kmarzecki.trollo.security

import com.kmarzecki.trollo.model.auth.LoginDto
import com.kmarzecki.trollo.model.auth.LoginResponse
import com.kmarzecki.trollo.model.auth.RegisterDto
import com.kmarzecki.trollo.model.auth.UserResponse
import java.security.Principal

/**
 * Authentication service
 */
interface AuthenticationService {
    /**
     * Login user
     * @param dto dto containing login information
     * @return logged user response
     */
    fun login(dto: LoginDto): LoginResponse

    /**
     * Register user
     * @param registerDto dto containing registration information
     */
    fun register(registerDto: RegisterDto)

    /**
     * Get user information
     * @param principal user principal
     * @return User information response
     */
    fun getMe(principal: Principal): UserResponse
}